api/v1/categories/PkpCategoryCategoryController.php

```diff
@@ -0,0 +1,409 @@
+        }
+
+        return response()->json([], Response::HTTP_OK);
+    }
+
+    /**
+     * Create a new category.
+     * Include `parentCategoryId` in query params to create new Category as a sub-category of an existing one.
+     */
+    public function add(Request $illuminateRequest): JsonResponse
+    {
+        return $this->saveCategory($illuminateRequest);
+    }
+
+    /**
+     * Edit an existing category by ID.
+     */
+    public function edit(Request $illuminateRequest): JsonResponse
+    {
+        return $this->saveCategory($illuminateRequest);
+    }
+
+    /**
+     * Create or update a category.
+     *
+     * Used internally to handle both new category creation and editing existing ones.
+     */
+    private function saveCategory(Request $illuminateRequest): JsonResponse
+    {
+        $context = $this->getRequest()->getContext();
+        $user = $this->getRequest()->getUser();
+        $parentId = $illuminateRequest->input('parentCategoryId') ? (int)$illuminateRequest->input('parentCategoryId') : null;
+        $categoryId = $illuminateRequest->route('categoryId') ? (int)$illuminateRequest->route('categoryId') : null;
+
+        if ($parentId) {
+            if (!Repo::category()->exists($parentId, $context->getId())) {
+                return response()->json(__('api.404.resourceNotFound'), Response::HTTP_NOT_FOUND);
+            }
+        }
+
+        // Get a category object to edit or create
+        if ($categoryId == null) {
+            $category = Repo::category()->dao->newDataObject();
+            $category->setContextId($context->getId());
+        } else {
+            $category = Repo::category()->get($categoryId, $context->getId());
+
+            if (!$category) {
+                return response()->json(__('api.404.resourceNotFound'), Response::HTTP_NOT_FOUND);
+            }
+
+            $parentId = $category->getParentId();
+        }
+
+        $params = $this->convertStringsToSchema(\PKP\services\PKPSchemaService::SCHEMA_CATEGORY, $illuminateRequest->input());
+        $params['contextId'] = $category->getContextId();
+        $errors = Repo::category()->validate($category, $params, $context);
+
+        if (!empty($errors)) {
+            return response()->json($errors, Response::HTTP_BAD_REQUEST);
+        }
+
+        // Set the editable properties of the category object
+        $category->setTitle($illuminateRequest->input('title'), null);
+        $category->setDescription($illuminateRequest->input('description'), null);
+        $category->setParentId($parentId);
+        $category->setPath($illuminateRequest->input('path'));
+        $category->setSortOption($illuminateRequest->input('sortOption'));
+
+        // Update or insert the category object
+        if ($categoryId == null) {
+            $category->setSequence(REALLY_BIG_NUMBER);
+            $categoryId = Repo::category()->add($category);
+            Repo::category()->dao->resequenceCategories($context->getId());
```

Comment:
Instead of using the `REALLY_BIG_NUMBER`, then resequencing (which will run another query/loop), you can just set the sequence with the right value (i.e. `max() + 1`).

Reply:
Given that we are removing the ability to order categories, resequencing can be removed as if no longer serves any purpose